source/reference/data-types.txt

```diff
@@ -38,325 +40,495 @@ string or as a ``Date`` object:
+explicitly specify a 64-bit integer.
 
 .. code-block:: javascript
 
-   var myDate = new Date();
-   var myDateInitUsingISODateWrapper = ISODate();
+      db.types.insertOne(
+                          {
+                              "_id": 3,
+                              "value": Long("1"),
+                              "expectedType": "Long" 
+                          }
+                        )
 
-You can use the ``new`` operator with the ``ISODate()`` constructor as
-well.
-
-To print the value of the variable, type the variable name in the
-shell, as in the following:
+.. note::
 
-.. code-block:: javascript
+   In the legacy :binary:`~bin.mongo` shell ``NumberLong()`` accepted
+   either a string or integer value. In ``mongosh``, ``NumberLong()``
+   only accepts string values. ``Long()`` accepts integers and strings.
 
-   myDate
+.. _shell-type-decimal:
 
-The result is the ``Date`` value of ``myDate`` wrapped in the
-``ISODate()`` helper:
+Decimal128
+----------
 
-.. code-block:: javascript
+:node-api:`Decimal128() <Decimal128.html>` values are  128-bit
+decimal-based floating-point numbers that emulate decimal rounding with
+exact precision. 
 
-   ISODate("2012-12-19T06:01:17.171Z")
+This functionality is intended for applications that handle
+:manual:`monetary data </tutorial/model-monetary-data>`, such as
+financial, tax, and scientific computations. 
 
-To verify the type, use the ``instanceof`` operator, as in the
-following:
+The ``Decimal128`` :manual:`BSON type </reference/bson-types>`
+uses the IEEE 754 decimal128 floating-point numbering format which
+supports 34 decimal digits (i.e. significant digits) and an exponent
+range of −6143 to +6144.
 
 .. code-block:: javascript
 
-   myDate instanceof Date
-   myDateInitUsingISODateWrapper instanceof Date
-
-The operation returns ``true`` for both.
+      db.types.insertOne(
+                          {
+                              "_id": 5,
+                              "value": Decimal128("1"),
+                              "expectedType": "Decimal128" 
+                          }
+                       )
 
-ObjectId
-~~~~~~~~
 
-``mongosh`` provides the ``ObjectId()`` wrapper class around the
-:ref:`objectid` data type. To generate a new ObjectId, use the following
-operation in ``mongosh``:
+.. note:: To use the ``Decimal128`` data type with a
+   :ecosystem:`MongoDB driver </drivers/>`, be sure to use a driver
+   version that supports it.
 
-.. code-block:: javascript
+Equality and Sort Order
+~~~~~~~~~~~~~~~~~~~~~~~
 
-   new ObjectId
+Values of the ``Decimal128`` type are compared and sorted with other
+numeric types based on their actual numeric value.  Numeric values
+of the binary-based ``Double`` type generally have approximate
+representations of decimal-based values and may not be exactly
+equal to their decimal representations. You can use the
+``Decimal128()`` constructor to check the equality of decimal values.
```

Comment:
Dumb question – how?

Reply:
The line was a summary of the prior copy. I tried making it work using some hokey wrapping, but it's probably better just to drop the statement.

Reply:
Yeah, I think it might be better to either tell users how to check for equality (I guess comparing the `.toString()` results?) or not mention it 👍